tooling/v8-snapshot/src/setup/install-snapshot.ts

```diff
@@ -0,0 +1,158 @@
+import path from 'path'
+import { SnapshotGenerator } from '../generator/snapshot-generator'
+import { prettyPrintError } from '../utils'
+import fs from 'fs-extra'
+
+const debug = require('debug')
+const logInfo = debug('cypress:snapgen:info')
+const logDebug = debug('cypress:snapgen:debug')
+
+/*
+ * Tries to resolve results from the previous step for the given environment.
+ * Returns empty object if resolution failed.
+ */
+function resolvePrevious ({ snapshotMetaPrevFile }) {
+  try {
+    const {
+      norewrite: previousNoRewrite,
+      deferred: previousDeferred,
+      healthy: previousHealthy,
+    } = require(snapshotMetaPrevFile)
+
+    return { previousNoRewrite, previousDeferred, previousHealthy }
+  } catch (_) {
+    return { previousNoRewrite: [], previousDeferred: [], previousHealthy: [] }
+  }
+}
+
+function getSnapshotGenerator ({
+  nodeModulesOnly,
+  projectBaseDir,
+  snapshotCacheDir,
+  snapshotEntryFile,
+  snapshotMetaPrevFile,
+  usePreviousSnapshotMetadata,
+  resolverMap,
+}: {
+  nodeModulesOnly: boolean
+  projectBaseDir: string
+  snapshotCacheDir: string
+  snapshotEntryFile: string
+  snapshotMetaPrevFile: string
+  usePreviousSnapshotMetadata: boolean
+  resolverMap: Record<string, string>
+}) {
+  const {
+    previousNoRewrite,
+    previousDeferred,
+    previousHealthy,
+  } = usePreviousSnapshotMetadata
+    ? resolvePrevious({
+      snapshotMetaPrevFile,
+    })
+    : {
+      previousNoRewrite: [], previousDeferred: [], previousHealthy: [],
+    }
+
+  return new SnapshotGenerator(projectBaseDir, snapshotEntryFile, {
+    cacheDir: snapshotCacheDir,
+    previousDeferred,
+    previousHealthy,
+    previousNoRewrite,
+    nodeModulesOnly,
+    resolverMap,
+    // sourcemapEmbed: true,
+    forceNoRewrite: [
+      'node_modules/@cypress/get-windows-proxy/src/registry.js',
```

Comment:
Document this a little better

Reply:
Updated: [`7cf1596` (#23732)](https://github.com/cypress-io/cypress/pull/23732/commits/7cf159636e2d08071df90e02c844ecb45dd26ba0)